ci/scripts/license_diff.py

```
@@ -0,0 +1,138 @@
+    """Resolve a package license from PyPI metadata.
+
+    Args:
+        name: Distribution name on PyPI.
+        version: Optional version pin used to query version-specific metadata.
+
+    Returns:
+        A best-effort license string from the available metadata fields.
+    """
+    # Use version-specific metadata when available to avoid mismatches.
+    url = f"https://pypi.org/pypi/{name}/json" if version is None else f"https://pypi.org/pypi/{name}/{version}/json"
+    with urllib.request.urlopen(url) as r:
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)